pkg/generate/app/cmd/newapp.go

```diff
@@ -418,6 +418,19 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						return nil, err
 					}
 				}
+				// Append any exposed ports from Dockerfile to input image
```

Comment:
Port detection was previously happening while in source detection (removed by [this PR](https://github.com/openshift/origin/commit/5ab34d377db4710589d0b3f15cebea23e1f6500f). We should restore it somewhere there and not so late in the new-app flow.